From e691803e130d1aced38c21e98c41ce40bda44ffe Mon Sep 17 00:00:00 2001
From: Hugues Kamba-Mpiana <hugues.kambampiana@arm.com>
Date: Tue, 19 Dec 2023 14:09:27 +0000
Subject: [PATCH 1/4] resources: Replace model wav2letter with tiny_wav2letter

`wav2letter` model needs singificant amount of memory (around 14Mb),
where as `tiny_wav2letter` needs around 4Mb.
Update the location of the resource where referenced for
the `asr` and `kws` use cases.

Signed-off-by: Hugues Kamba-Mpiana <hugues.kambampiana@arm.com>
---
 set_up_default_resources.py           | 16 ++++++++--------
 source/use_case/asr/usecase.cmake     |  4 ++--
 source/use_case/kws_asr/usecase.cmake |  4 ++--
 3 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/set_up_default_resources.py b/set_up_default_resources.py
index b442432..2cb987d 100755
--- a/set_up_default_resources.py
+++ b/set_up_default_resources.py
@@ -51,16 +51,16 @@ json_uc_res = [
     {
         "use_case_name": "asr",
         "url_prefix": [
-            "https://github.com/ARM-software/ML-zoo/raw/1a92aa08c0de49a7304e0a7f3f59df6f4fd33ac8/models/speech_recognition/wav2letter/tflite_pruned_int8/"
+            "https://github.com/ARM-software/ML-zoo/raw/eb2170aac1317f00b128ab82bdb159cfcca36ea6/models/speech_recognition/tiny_wav2letter/tflite_pruned_int8/"
         ],
         "resources": [
             {
-                "name": "wav2letter_pruned_int8.tflite",
-                "url": "{url_prefix:0}wav2letter_pruned_int8.tflite",
+                "name": "tiny_wav2letter_pruned_int8.tflite",
+                "url": "{url_prefix:0}tiny_wav2letter_pruned_int8.tflite",
             },
             {
                 "name": "ifm0.npy",
-                "url": "{url_prefix:0}testing_input/input_2_int8/0.npy",
+                "url": "{url_prefix:0}testing_input/input_1_int8/0.npy",
             },
             {
                 "name": "ofm0.npy",
@@ -131,18 +131,18 @@ json_uc_res = [
     {
         "use_case_name": "kws_asr",
         "url_prefix": [
-            "https://github.com/ARM-software/ML-zoo/raw/1a92aa08c0de49a7304e0a7f3f59df6f4fd33ac8/models/speech_recognition/wav2letter/tflite_pruned_int8/",
+            "https://github.com/ARM-software/ML-zoo/raw/eb2170aac1317f00b128ab82bdb159cfcca36ea6/models/speech_recognition/tiny_wav2letter/tflite_pruned_int8/",
             "https://github.com/ARM-software/ML-zoo/raw/9f506fe52b39df545f0e6c5ff9223f671bc5ae00/models/keyword_spotting/micronet_medium/tflite_int8/",
         ],
         "resources": [
             {
-                "name": "wav2letter_pruned_int8.tflite",
-                "url": "{url_prefix:0}wav2letter_pruned_int8.tflite",
+                "name": "tiny_wav2letter_pruned_int8.tflite",
+                "url": "{url_prefix:0}tiny_wav2letter_pruned_int8.tflite",
             },
             {
                 "sub_folder": "asr",
                 "name": "ifm0.npy",
-                "url": "{url_prefix:0}testing_input/input_2_int8/0.npy",
+                "url": "{url_prefix:0}testing_input/input_1_int8/0.npy",
             },
             {
                 "sub_folder": "asr",
diff --git a/source/use_case/asr/usecase.cmake b/source/use_case/asr/usecase.cmake
index fc3f2b5..3ce2c0a 100644
--- a/source/use_case/asr/usecase.cmake
+++ b/source/use_case/asr/usecase.cmake
@@ -75,9 +75,9 @@ USER_OPTION(${use_case}_ACTIVATION_BUF_SZ "Activation buffer size for the chosen
     STRING)
 
 if (ETHOS_U_NPU_ENABLED)
-    set(DEFAULT_MODEL_PATH      ${DEFAULT_MODEL_DIR}/wav2letter_pruned_int8_vela_${ETHOS_U_NPU_CONFIG_ID}.tflite)
+    set(DEFAULT_MODEL_PATH      ${DEFAULT_MODEL_DIR}/tiny_wav2letter_pruned_int8_vela_${ETHOS_U_NPU_CONFIG_ID}.tflite)
 else()
-    set(DEFAULT_MODEL_PATH      ${DEFAULT_MODEL_DIR}/wav2letter_pruned_int8.tflite)
+    set(DEFAULT_MODEL_PATH      ${DEFAULT_MODEL_DIR}/tiny_wav2letter_pruned_int8.tflite)
 endif()
 
 set(EXTRA_MODEL_CODE
diff --git a/source/use_case/kws_asr/usecase.cmake b/source/use_case/kws_asr/usecase.cmake
index de84bfd..9bfbbd0 100644
--- a/source/use_case/kws_asr/usecase.cmake
+++ b/source/use_case/kws_asr/usecase.cmake
@@ -68,10 +68,10 @@ USER_OPTION(${use_case}_MODEL_SCORE_THRESHOLD_ASR "Specify the score threshold [
 
 if (ETHOS_U_NPU_ENABLED)
     set(DEFAULT_MODEL_PATH_KWS      ${DEFAULT_MODEL_DIR}/kws_micronet_m_vela_${ETHOS_U_NPU_CONFIG_ID}.tflite)
-    set(DEFAULT_MODEL_PATH_ASR      ${DEFAULT_MODEL_DIR}/wav2letter_pruned_int8_vela_${ETHOS_U_NPU_CONFIG_ID}.tflite)
+    set(DEFAULT_MODEL_PATH_ASR      ${DEFAULT_MODEL_DIR}/tiny_wav2letter_pruned_int8_vela_${ETHOS_U_NPU_CONFIG_ID}.tflite)
 else()
     set(DEFAULT_MODEL_PATH_KWS      ${DEFAULT_MODEL_DIR}/kws_micronet_m.tflite)
-    set(DEFAULT_MODEL_PATH_ASR      ${DEFAULT_MODEL_DIR}/wav2letter_pruned_int8.tflite)
+    set(DEFAULT_MODEL_PATH_ASR      ${DEFAULT_MODEL_DIR}/tiny_wav2letter_pruned_int8.tflite)
 endif()
 
 USER_OPTION(${use_case}_MODEL_TFLITE_PATH_KWS "NN models file to be used for KWS in the evaluation application. Model files must be in tflite format."
-- 
2.40.1

